// @atlaskit/tokens/dist/cjs/palettes/spacing-scale.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var scale = {
  spacing: {
    scale: {
      Space0: {
        value: '0',
        attributes: {
          group: 'scale'
        }
      },
      Space025: {
        value: '0.125rem',
        attributes: {
          group: 'scale'
        }
      },
      Space050: {
        value: '0.25rem',
        attributes: {
          group: 'scale'
        }
      },
      Space075: {
        value: '0.375rem',
        attributes: {
          group: 'scale'
        }
      },
      Space100: {
        value: '0.5rem',
        attributes: {
          group: 'scale'
        }
      },
      Space150: {
        value: '0.75rem',
        attributes: {
          group: 'scale'
        }
      },
      Space200: {
        value: '1rem',
        attributes: {
          group: 'scale'
        }
      },
      Space250: {
        value: '1.25rem',
        attributes: {
          group: 'scale'
        }
      },
      Space300: {
        value: '1.5rem',
        attributes: {
          group: 'scale'
        }
      },
      Space400: {
        value: '2rem',
        attributes: {
          group: 'scale'
        }
      },
      Space500: {
        value: '2.5rem',
        attributes: {
          group: 'scale'
        }
      },
      Space600: {
        value: '3rem',
        attributes: {
          group: 'scale'
        }
      }
    }
  }
};
var _default = scale;
exports.default = _default;
